### PR TITLE
kv/union_store:remove tableinfo from union_store

### PR DESCRIPTION
**Pull request by AndreMouche**
Signed-off-by: AndreMouche

### What problem does this PR solve?
This PR removes tableinfo from union_store
Part of https://github.com/pingcap/tidb/issues/22513

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

**Review by disksing — APPROVED** [review]

File: store/tikv/prewrite.go [hunk 1]
```diff
@@ -248,8 +248,8 @@ func (action actionPrewrite) handleSingleBatch(c *twoPhaseCommitter, bo *Backoff
 		for _, keyErr := range keyErrs {
 			// Check already exists error
 			if alreadyExist := keyErr.GetAlreadyExist(); alreadyExist != nil {
-				key := alreadyExist.GetKey()
-				return c.extractKeyExistsErr(key)
+				e := &ErrKeyExist{AlreadyExist: alreadyExist}
+				return c.extractKeyExistsErr(e)
```
Comment (sticnarf) [inline store/tikv/prewrite.go:252]: Is it true that after the change, the `AlreadyExist` error would have a less user-friendly message?

**Review by sticnarf — APPROVED** [review]

**Pull request merged**